src/cgame/default/cg_media.c

```diff
@@ -154,23 +154,18 @@ static void Cg_InitFootsteps(void) {
 
 	default_samples = g_array_append_vals(default_samples, (const s_sample_t *[]) {
 		cgi.LoadSample("#players/common/step_1"),
-		               cgi.LoadSample("#players/common/step_2"),
-		               cgi.LoadSample("#players/common/step_3"),
-		               cgi.LoadSample("#players/common/step_4")
+		cgi.LoadSample("#players/common/step_2"),
+		cgi.LoadSample("#players/common/step_3"),
+		cgi.LoadSample("#players/common/step_4")
 	}, 4);
 
 	g_hash_table_insert(cg_footstep_table, "default", default_samples);
-	
-	size_t num_materials = 0;
-	const cm_material_t **materials = cgi.MapMaterials(&num_materials);
 
-	if (num_materials) {
-		for (size_t i = 0; i < num_materials; i++) {
-			const cm_material_t *material = materials[i];
+	r_material_t **material = cgi.WorldModel()->materials;
```

Comment:
Curious, why do we pass the renderer-specific r_material_t to the cgame? It doesn't/shouldn't need to know about the abstraction, it only needs the cm_material_t pointers. Since r_material_t is a 'subclass' it should be fine to pass them back as cm_material_t; then our API won't suddenly be out of date because of r_material_t changing if we do changes in the renderer.

Ideally we'd do something similar for most of our renderer structs, keep them abstract as possible to the cgame.